Adds primvar interpolation example asset

ablev-nv opened the pull request:
One of the most common recurring questions I help with is about primvar interpolation. I hope this example asset and associated docs can help de-mystify the concepts.

As noted in the docs, this asset was inspired by / plagiarized from a post @jcowles made to usd-interest a few years ago. So, thank you / apologizes to Jeremy!

jcowles commented on test_assets/PrimvarInterpolation/README.md:
```diff
@@ -0,0 +1,221 @@
+
+The example file [primvar_interpolation.usda](primvar_interpolation.usda) was used to generated the screenshot in the overview of this document, and contains all of the examples discussed below.
+
+## constant
```
Nit picky, but would be nice to keep all the headings consistently capitalized

jcowles commented on test_assets/PrimvarInterpolation/README.md:
```diff
@@ -0,0 +1,221 @@
+
+# Mesh construction
+
+It will be helpful to review how mesh geometry is defined in USD. For this example, we define a mesh with two faces:
```
It might be good to mention that these different interpolation modes behave slightly differently on the type of geometry - e.g. uniform means something subtly different for BasisCurve vs. Mesh

Might also be good to call out "future work" of providing a basis curves example of how primvars interpolate, I think the [USD BasisCurves doc](https://graphics.pixar.com/usd/dev/api/class_usd_geom_basis_curves.html#UsdGeomBasisCurves_PrimvarInterpolation) is pretty difficult to understand today.

jcowles commented on test_assets/PrimvarInterpolation/README.md:
```diff
@@ -0,0 +1,221 @@
+# Primvar Interpolation
+
+![annotated primvar interpolation screenshot](images/annotated_results_storm.png)
```
The images are not showing up for me in the "display rich text" mode, not sure if they will work once the PR is merged.

AlexSchwank replied:
You can see how it'll look like when you click the "..." button on the top right and then "View File". It'll go to [this link](https://github.com/usd-wg/assets/blob/0d572735561950c44870d2706f3489f53dc0bc76/test_assets/PrimvarInterpolation/README.md).

jcowles commented on test_assets/PrimvarInterpolation/README.md:
```diff
@@ -0,0 +1,221 @@
+
+**Size of primvar array**
+
+number of faces on the mesh, which is the same as the length of **faceVertexCounts** array
```
Would be nice to keep all descriptive text grammatically correct (full sentences, start with capital, end with a period, etc)

Also, really great to explicitly call out the connection to faceVertexCounts

jcowles commented on test_assets/PrimvarInterpolation/README.md:
```diff
@@ -0,0 +1,221 @@
+
+**Notes**
+
+With constant interpolation, a single value is specified for the entire prim. It is important to note, this is still specified as an array, even though it will only have a single entry.
```
I don't know if you should make this super explicit, but it might be worth mentioning once:

The primvar is declared as color3f[] (which is a tuple of three values), so a single constant primvar value is an array of one tuple of three values.

```color3f[] primvars:displayColor = [(1, 0.3, 0.3)]```

jcowles commented on test_assets/PrimvarInterpolation/README.md:
````diff
@@ -0,0 +1,221 @@
+
+```
+int[] faceVertexCounts = [4, 4]
+# int[] faceVertexCounts = [A, B]
````
I really like this, because if the reader skips the descriptive text, I think the code will still be clear.

jcowles commented on test_assets/PrimvarInterpolation/README.md:
```diff
@@ -0,0 +1,221 @@
+
+**Size of primvar array**
+
+number of vertices on the mesh, which is the same as the length of **points** array
```
if you make the one above a complete sentence, this one should be too (and below)

jcowles commented on test_assets/PrimvarInterpolation/README.md:
````diff
@@ -0,0 +1,221 @@
+#)
+```
+
+## varying
````
Ditto for capitalization (and headings below)

jcowles commented on test_assets/PrimvarInterpolation/README.md:
```diff
@@ -0,0 +1,221 @@
+
+**Notes**
+
+With varying interpolation, one value is specified for each vertex of the mesh. The order of the array matches the order of the **points** array:
```
Might be worth calling out that this is known as "vertex colors" in most game engines.

jcowles commented on test_assets/PrimvarInterpolation/README.md:
````diff
@@ -0,0 +1,221 @@
+#)
+```
+
+You probably noticed that **varying** and **vertex** interpolation are quite similar. The differences between them are subtle, and relate to how they interact with subdivision on the mesh. You can see some differences in the example below, where the Mesh prims have a duplicate to their side that sets `token interpolateBoundary = "edgeOnly"`, which causes the corners of the mesh to be rounded when ...
````
I think it's important to call out here that for DCC tools exporting data, if the exported data is expected to be a polygon mesh (and the surface is explicitly not intended to be subdivided), then primvar interpolation should be set to "varying" and the mesh subdivision scheme should be set to "none", to ensure the mesh renders as intended when downstream consumers DO support subdivision.

As I'm writing this, I'm realizing there's a whole discussion here about how to handle data for export if it originates from the tool or if it came from an existing USD file, and might be out of scope for this...

jcowles reviewed (approved):
This is super awesome, the effort & thought you put into it shows!

I left several comments, I'm curious to hear what you (or others) think

meshula commented:
@ablev-nv I think with a couple of minor edits we're ready to merge?

ablev-nv commented:
Addressed comments here. I added some text that notes this asset only addresses primvar interpolation with Mesh prims. Interpolation with other types (BasisCurves) would be a good subsequent pull request, along with a more in-depth exploration of interpolation with respect to subdivision.

jcowles reviewed (approved):
Awesome, thank you!!
